agimus_controller/tests/test_mpc_data.py

```diff
@@ -0,0 +1,88 @@
+import numpy as np
+import random
+import unittest
+
+
+from agimus_controller.mpc_data import MPCDebugData, OCPDebugData, OCPResults
+
+
+class TestOCPParamsCrocoBase(unittest.TestCase):
+    """
+    TestOCPParamsCrocoBase unittests parameters settters and getters of OCPParamsBaseCroco class.
+    """
+
+    def __init__(self, methodName="runTest"):
+        super().__init__(methodName)
+
+    def test_ocp_results(self):
+        """
+        Test the initialization of the MPCDebugData class.
+        """
+        N = random.randint(10, 100)
+        rows = random.randint(10, 20)
+        cols = 2 * rows
+        params = {
+            "states": [np.random.rand(cols) for _ in range(N)],
+            "ricatti_gains": [np.random.rand(rows, cols) for _ in range(N)],
+            "feed_forward_terms": [np.random.rand(rows) for _ in range(N)],
+        }
+        obj = OCPResults(**params)
```

Comment:
```suggestion
        obj = OCPResults(**deepcopy(params))
```
This way we test that values are the same and not references